### PR TITLE
AIP-84: Migrating delete queued asset events for DAG to fastAPI

### PR DESCRIPTION
**Pull request by amoghrajesh**
related: https://github.com/apache/airflow/issues/42370
Migrating delete queued asset events for DAG to fastAPI 

Dependent on https://github.com/apache/airflow/pull/43934

Same setup as https://github.com/apache/airflow/pull/43934

Responses:
1. Legacy
image

2. FastAPI
image

With time filtering
1. Legacy
image

2. FastAPI
image

Time filtering but no queued event found
1. Legacy
image

2. FastAPI
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
Only the last 2 commits are relevant

**Review by pierrejeambrun — COMMENTED** [review]
Most of the small suggestions from https://github.com/apache/airflow/pull/43934 also apply here.

Otherwise looking good.

File: tests/api_fastapi/core_api/routes/public/test_dags.py [hunk 1]
```diff
@@ -557,3 +561,101 @@ def test_delete_dag(
...
+            f"/public/dags/{dag_id}/assets/queuedEvent",
+        )
+
+        assert response.status_code == 404
+        assert response.json()["detail"] == "Queue event with dag_id: `not_exists` was not found"
+
+
+class TestDeleteDagDatasetQueuedEvents(TestQueuedEventEndpoint):
+    @pytest.mark.usefixtures("time_freezer")
+    def test_should_respond_200(self, test_client, session, create_dummy_dag):
```
Comment (vatsrahul1001) [inline tests/api_fastapi/core_api/routes/public/test_dags.py]: Should be `test_should_respond_204`

**Comment by amoghrajesh** [comment]
Closing in favour of https://github.com/apache/airflow/pull/44129

**Pull request closed without merging**